primitives/core/src/sr25519.rs

```diff
@@ -530,8 +561,15 @@ impl TraitPair for Pair {
 	}
 
 	/// Verify a signature on a message. Returns true if the signature is good.
-	fn verify<M: AsRef<[u8]>>(sig: &Self::Signature, message: M, pubkey: &Self::Public) -> bool {
-		Self::verify_weak(&sig.0[..], message, pubkey)
+	fn verify_format<M: AsRef<[u8]>>(format: Self::Format, sig: &Self::Signature, message: M, pubkey: &Self::Public) -> bool {
```

Comment:
I think we should not do this. We should really prevent that this type by default can verify an old signature.

I think we should only support checking the old signature in the implementation of the old `sr25519_verify` function.

Reply:
By default it will deny old signature, see `Format::default` returns `Format::V2`

Reply:
Yeah, you are right! Too much versioning 😅